hq/app/views/gcp/gcpReportBody.scala.html

```diff
@@ -28,10 +28,27 @@
                         <td class="gcp-table-row-severity"><span>@finding.severity</span></td>
                         <td class="gcp-table-row-category"><span>@finding.category</span></td>
                         <td class="gcp-table-row-date"><span>@finding.eventTime.toString(GcpDisplay.dateTimePattern)</span></td>
-                        <td><span>@finding.explanation</span></td>
-                        <td><span>@finding.recommendation</span></td>
+                        <td>
+                            @if(finding.explanation.getOrElse("").length < 30) {
+                                <span class="unaltered-short">@finding.explanation</span>} else {
+                                <span class="gcp-toggle gcp-truncated">@GcpDisplay.preview(finding.explanation.getOrElse(""), 30)</span>
+                                <span class="gcp-toggle gcp-full-text">@finding.explanation</span>
+                                <button class="js-read-more mdc-button">
+                                    <span class="mdc-button__label">Read More/Less</span>
+                                </button>
+                            }
+                        </td>
+                        <td>
+                            @if(finding.recommendation.getOrElse("").length < 30) {
```

Comment:
you can do finding.recommendation.exists(_.length < 30)  I think

Reply:
or alternatively wrap this entire eblock in a `@finding.recommendatiion.map{ r +> ` to avoid the other .getOrElse